pkg/server/providertargets/service_test.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package providertargets_test
+
+import (
+	"testing"
+
+	"github.com/daytonaio/daytona/internal/testing/provider/targets"
+	"github.com/daytonaio/daytona/pkg/provider"
+	"github.com/daytonaio/daytona/pkg/server/providertargets"
+	"github.com/stretchr/testify/suite"
+)
+
+var providerTarget1 *provider.ProviderTarget = &provider.ProviderTarget{
+	Name: "target1",
+	ProviderInfo: provider.ProviderInfo{
+		Name:    "provider1",
+		Version: "v1",
+	},
+	Options: "",
+}
+
+var providerTarget2 *provider.ProviderTarget = &provider.ProviderTarget{
+	Name: "target2",
+	ProviderInfo: provider.ProviderInfo{
+		Name:    "provider2",
+		Version: "v1",
+	},
+	Options: "",
+}
+
+var providerTarget3 *provider.ProviderTarget = &provider.ProviderTarget{
+	Name: "target3",
+	ProviderInfo: provider.ProviderInfo{
+		Name:    "provider1",
+		Version: "v1",
+	},
+	Options: "",
+}
+
+var providerTarget4 *provider.ProviderTarget = &provider.ProviderTarget{
+	Name: "new-target",
+	ProviderInfo: provider.ProviderInfo{
+		Name:    "provider2",
+		Version: "v1",
+	},
+	Options: "",
+}
+
+var expectedProviderTargets []*provider.ProviderTarget
+var expectedProviderTargetsMap map[string]*provider.ProviderTarget
+
+type ProviderTargetServiceTestSuite struct {
+	suite.Suite
+	providerTargetService providertargets.IProviderTargetService
+	targetStore           provider.TargetStore
+}
+
+func NewProviderTargetServiceTestSuite() *ProviderTargetServiceTestSuite {
+	return &ProviderTargetServiceTestSuite{}
+}
+
+func (s *ProviderTargetServiceTestSuite) SetupTest() {
+	expectedProviderTargets = []*provider.ProviderTarget{
+		providerTarget1, providerTarget2, providerTarget3,
+	}
+
+	expectedProviderTargetsMap = map[string]*provider.ProviderTarget{
+		providerTarget1.Name: providerTarget1,
+		providerTarget2.Name: providerTarget2,
+		providerTarget3.Name: providerTarget3,
+	}
+
+	s.targetStore = targets.NewInMemoryTargetStore()
+	s.providerTargetService = providertargets.NewProviderTargetService(providertargets.ProviderTargetServiceConfig{
+		TargetStore: s.targetStore,
+	})
+
+	for _, target := range expectedProviderTargets {
+		_ = s.providerTargetService.Save(target)
+	}
+}
+
+func (s *ProviderTargetServiceTestSuite) BeforeTest(_, testName string) {
```

Comment:
[The comment](https://github.com/daytonaio/daytona/pull/435#discussion_r1580563331) from the apikey tests applies to this function as well.